plasmapy/tests/helpers/expected.py

```diff
@@ -0,0 +1,198 @@
+"""Contains a class to represent the expected outcome of a test."""
+
+__all__ = ["ExpectedTestOutcome"]
+
+import inspect
+
+from typing import Any
+
+from plasmapy.utils.code_repr import call_string
+
+
+def _is_warning(obj) -> bool:
+    """Return `True` if the argument is a warning, and `False` otherwise."""
+    return inspect.isclass(obj) and issubclass(obj, Warning)
+
+
+def _is_exception(obj) -> bool:
+    """Return `True` if the argument is an exception, and `False` otherwise."""
+    if not inspect.isclass(obj):
+        return False
+    else:
+        return issubclass(obj, BaseException) and not issubclass(obj, Warning)
```

Comment:
You'd have to have some really messed up design to have a warning that derives from BaseException, no? :sweat_smile: